merge-all.rb

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env ruby
+
+# Usage:
+# "REPOS_PATH=infrastructure GH_ACCESS_TOKEN=abc123 ./merge-all.rb"
+BRANCH_NAME = 'update-dependencies'
+
+def ruby_repos_file
+  File.open("#{ENV['REPOS_PATH']}/ruby")
+end
+
+def repos
+  [ruby_repos_file].map(&:to_a).flatten.map(&:strip).sort
+end
+
+# @return [Array<Hash>] the update PR
+def find_prs(client, repos)
+  repo_content = repos.map do |repo|
+    pr, * = client.pull_requests(repo, head: "sul-dlss:#{BRANCH_NAME}")
+
+    unless pr
+      warn "no #{BRANCH_NAME} pr found for #{repo}"
+      next
+    end
+    statuses = client.combined_status(repo, pr.head.sha)
+
+
+    branch_protection = client.branch_protection(repo, pr.base.ref, accept: Octokit::Preview::PREVIEW_TYPES[:branch_protection])
+    warn "No branch branch_protection is set up for #{repo}" unless branch_protection
+
+    # GitHub API marks PRs with 0 statuses as "pending", we cast that to success
+    status = statuses.total_count == 0 ? 'success' : statuses.state
+    { repo: repo, number: pr.number, url: pr.html_url, status: status }
+  end.compact
+end
+
+def access_token
+  ENV['GH_ACCESS_TOKEN']
+end
+
+require 'octokit'
+client = Octokit::Client.new(access_token: access_token)
+pr_list = find_prs(client, repos)
+
+
+if pr_list.empty?
+  puts "No PRs were found"
+  exit
+end
+
+unless pr_list.all? { |pr| pr[:status] == 'success' }
+  puts "These PRs are not passing: "
```

Comment:
@ndushay Instead of, or in addition to, the comment you requested above, how about amending this `puts` line to make clear to the runner (who may not be reading this code) that no PRs were merged? Something like:

```ruby
  puts '*No* PRs were merged because these PRs are not passing: '
```

Reply:
I like this idea a lot.

Reply:
@ndushay I added it as a suggestion for easier inclusion. cc: @jcoyne